packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1679,14 +1699,46 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         }
     }
 
-    public async createDataStore(pkg: string | string[]): Promise<IFluidRouter> {
-        return this._createDataStore(pkg, false /* isRoot */);
+    private async createDataStoreCore(
```

Comment:
It's a bit hard to follow the intention here. Any chance you would add comments above each function on what's the purpose of each of them and when they are expected to be called?
For example, I'd think (from one usage example) that the better name for this is createDataStoreLegacy(), to stress that it is used in scenarios not involving aliasing path. But it is used on aliasing path, which is a bit confusing, because it already assigned global name to this data store.

Also it would be great for APIs not to have both isRoot & id. specifying ID means it's root, and vice versa (even if it was already broken from this angle)